src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -813,12 +813,17 @@ UpdateCheckResult SotaUptaneClient::checkUpdates() {
   unsigned int ecus_count = 0;
   if (!uptaneOfflineIteration(&updates, &ecus_count)) {
     LOG_ERROR << "Invalid UPTANE metadata in storage.";
-    result = UpdateCheckResult({}, 0, UpdateStatus::kError, "Invalid UPTANE metadata in storage.");
+    result = UpdateCheckResult({}, 0, UpdateStatus::kError, Json::nullValue, "Invalid UPTANE metadata in storage.");
   } else {
+    std::string director_targets;
+    storage->loadNonRoot(&director_targets, Uptane::RepositoryType::Director, Uptane::Role::Targets());
```

Comment:
Should we check the return value here?

Reply:
If `loadNonRoot()` returns `false`, `director_targets` will be an empty string and `parseJSON()` will return `Json::Nullvalue` which will be passed as targets metadata. It looks like a reasonable error handling to me. Could we do something better?

Reply:
I'm just wondering if it's still OK to return UpdatesAvailable status in this case.

Reply:
It's more a style question to me. If `uptaneOfflineIteration` returned `true`, there should be some targets in the storage, so this check will test what our own code does (which is a questionable style to me). And even if it would be possible that `uptaneOfflineIteration` succeeds, but there is no targets metadata in the storage, I don't think it's a reason to fail, because our default solution ignores targets metadata in `UpdateCheckResults`.

Reply:
OK, makes sense to me. Thanks for the explanation!